langchain/src/chains/api/api_chain.ts

```diff
@@ -64,16 +70,46 @@ export class APIChain extends BaseChain implements APIChainInput {
   ): Promise<ChainValues> {
     const question: string = values[this.inputKey];
 
-    const api_url = await this.apiRequestChain.predict(
+    const api_json = await this.apiRequestChain.predict(
       { question, api_docs: this.apiDocs },
       runManager?.getChild()
     );
 
-    const res = await fetch(api_url, { headers: this.headers });
+    let api_options;
+    try {
+      api_options = await APIChain.getApiParser().parse(api_json);
+    } catch (e) {
+      const fixParser = OutputFixingParser.fromLLM(
+        this.llm,
+        APIChain.getApiParser()
+      );
+      api_options = await fixParser.parse(api_json);
+    }
+
+    const request_options =
```

Comment:
I think it would be nice to actually have this configurable - basically pass in a list of `allowedMethods` that defaults to `GET` and `POST`. 

Just get a little bit nervous now that I see DELETE here in code.

Reply:
Sure will do that. Makes sense @jacoblee93 !